### DISCUSSION TITLE
Any way to run `clippy` not only on save?

### DISCUSSION THREAD

**Discussion by LennyLizowzskiy** [mrcjkb/rustaceanvim#334] [Q&A] [answered]
### Feature description

Is there a way to run "flychecks" on demand (at least in Normal mode)?
I don't see any mention of it in the documentation

Related rust-analyzer PR and rust-lang issue: https://github.com/rust-lang/rust-analyzer/issues/13125, https://github.com/rust-lang/rust-analyzer/pull/13785

**Comment by mrcjkb** [comment, marked as answer]
Yes, see "fly check" in the readme's [usage section](https://github.com/mrcjkb/rustaceanvim?tab=readme-ov-file#usage).

**Reply by LennyLizowzskiy** [reply]
Sorry, didn't see it

**Reply by LennyLizowzskiy** [reply]
@mrcjkb 
Hi again.

`RustLsp flyCheck` doesn't seem to work

I ran it via this autocmd and also tried to run the command myself
```lua
autocmd({ "LspAttach", "FileAppendPost", "TextChangedI" }, {
        pattern = { "rust" },
        command = "RustLsp flyCheck run",
      })
```

File itself
![image](https://github.com/mrcjkb/rustaceanvim/assets/46971551/baeacaf1-a925-419d-8c7c-b2315ed66c4b)

Commented out these lines but diagnostics are still there
![image](https://github.com/mrcjkb/rustaceanvim/assets/46971551/0188b8d4-938c-4c75-a76e-94657adc2edd)

Even though it says that it ran clippy
![image](https://github.com/mrcjkb/rustaceanvim/assets/46971551/91825dcf-f6f1-46e7-90e2-315fd1470b57)

Correct changes in diagnostics appear only when I save the file via `:w`. Lsp notifies that it ran clippy once again, but this time it works
![image](https://github.com/mrcjkb/rustaceanvim/assets/46971551/186a2409-32eb-401b-9ae4-c8bb06f16752)

**Reply by mrcjkb** [reply]
You still need to save the file for cargo or clippy to be able to check it.
flycheck is only useful if you have `checkOnSave` disabled in your rust-analyzer settings, so that it doesn't check every time you save (e.g. if doing so is costly).

Afaik, rust-analyzer currently doesn't support checking unsaved buffers.
